ingress-examples/autoscale/README.md

```diff
@@ -0,0 +1,124 @@
+
+The ALBs expose [various metrics](https://kubernetes.github.io/ingress-nginx/user-guide/monitoring/#exposed-metrics) including request statistics and nginx process metrics. These metrics can be captured and aggregated by a metric collector, such as [Prometheus](https://prometheus.io/docs/introduction/overview/) and made available to Kubernetes HPA using [Prometheus Adapter](https://github.com/kubernetes-sigs/prometheus-adapter).
+
+Depending on your use case, you might want to scale your ALBs based on custom metrics, for example based on the number of incoming requests per second or the number of established connections. In the followings we present an example on setting up autoscaling based on custom metrics. With this example you can get started easily on designing your custom metrics based setup. Please note that the configuration in this example is not a production ready setup, but a simple deployment for demonstration purposes.
+
+For more information about autoscaling, checkout our [official documentation](https://cloud.ibm.com/docs/containers?topic=containers-ingress-alb-manage#alb_replicas_autoscaler).
+
+## Setup
+
+1. Replace the `<editme>` with your host and secret name in the `host`, `hosts` and `secretName` fields in the `example-deployment/example-ingress.yaml` file.
+
+2. Now you can deploy an example application with ingress, Prometheus, and Prometheus adapter:
+    * create `alb-autoscale-example` namespace for the new application with `kubectl apply -f alb-autoscale-example/autoscale-example-namespace.yaml` command. The new namespace will appear immediately:
+
+    ```
+    $ kubectl get ns alb-autoscale-example
+    NAME                    STATUS   AGE
+    alb-autoscale-example   Active   13s
+    ```
+
+    * deploy Promethus with `kubectl apply --kustomize alb-autoscale-example/prometheus` command. This will create a `prometheus-server` deployment in `alb-autoscale-example` namespace:
+
+    ```
+    $ kubectl get deployment -n alb-autoscale-example prometheus-server
+    NAME                READY   UP-TO-DATE   AVAILABLE   AGE
+    prometheus-server   1/1     1            1           30s
+    ```
+
+    * deploy Prometheus Adapter with `kubectl apply --kustomize alb-autoscale-example/prometheus-adapter` command. This will create a `alb-prometheus-adapter` deployment in `alb-autoscale-example` namespace. Note: the `alb-autoscale-example/prometheus-adapter/configmap.yaml` contains the description of the `nginx_ingress_controller_requests_rate` custom metric.
+
+    ```
+    $ kubectl get deployment -n alb-autoscale-example alb-prometheus-adapter
+    NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
+    alb-prometheus-adapter   1/1     1            1           50s
+    ```
+
+    * deploy the test application with `kubectl apply --kustomize alb-autoscale-example/example-deployment` command. This will create a `alb-autoscale-example` deployment in `alb-autoscale-example` namespace:
```

Comment:
```suggestion
    * Deploy the test application with `kubectl apply --kustomize alb-autoscale-example/example-deployment` command. This will create a `alb-autoscale-example` deployment in `alb-autoscale-example` namespace:
```